wkcuber/convert_nifti.py

```diff
@@ -72,129 +106,200 @@ def to_target_datatype(data: np.ndarray, target_dtype) -> np.ndarray:
 
 
 def convert_nifti(
-    source_nifti_path, target_path, layer_name, dtype, scale, mag=1, file_len=256
+    source_nifti_path,
+    target_path,
+    layer_name,
+    dtype,
+    scale,
+    mag=1,
+    file_len=DEFAULT_WKW_FILE_LEN,
+    bbox_to_enforce=None,
+    write_tiff=False,
+    use_orientation_header=False,
 ):
-    target_wkw_info = WkwDatasetInfo(
-        str(target_path.resolve()),
-        layer_name,
-        mag,
-        wkw.Header(
-            np.dtype(dtype),
-            block_type=wkw.Header.BLOCK_TYPE_LZ4HC,
-            file_len=file_len // 32,
-        ),
-    )
-    ensure_wkw(target_wkw_info)
-
+    voxels_per_cube = file_len * DEFAULT_WKW_VOXELS_PER_BLOCK
     ref_time = time.time()
     # Assume no translation
     offset = (0, 0, 0)
 
-    with open_wkw(target_wkw_info) as target_wkw:
-        source_nifti = nib.load(str(source_nifti_path.resolve()))
-        cube_data = np.array(source_nifti.get_fdata())
+    source_nifti = nib.load(str(source_nifti_path.resolve()))
 
-        if len(source_nifti.shape) == 3:
-            size = list(source_nifti.shape)
-            cube_data = cube_data.reshape((1,) + source_nifti.shape)
+    if use_orientation_header:
+        # Get canonical representation of data to incorporate
+        # encoded transformations. Needs to be flipped later
+        # to match the coordinate system of WKW.
+        source_nifti = nib.funcs.as_closest_canonical(source_nifti, enforce_diag=False)
 
-        elif len(source_nifti.shape) == 4:
-            size = list(source_nifti.shape[:-1])
-            cube_data = np.transpose(cube_data, (3, 0, 1, 2))
+    cube_data = np.array(source_nifti.get_fdata())
 
-        else:
-            logging.warning(
-                "Converting of {} failed! Too many or too less dimensions".format(
-                    source_nifti_path
-                )
+    is_probably_binary = np.unique(cube_data).shape[0] == 2
+    assume_segmentation_layer = False  # Since webKnossos does not support multiple segmention layers, this is hardcoded to False right now.
+
+    max_cell_id_args = (
+        {"largest_segment_id": int(np.max(cube_data) + 1)}
+        if assume_segmentation_layer
+        else {}
+    )
+    category_type = "segmentation" if assume_segmentation_layer else "color"
```

Comment:
Just to clarify, this means all Niftis will be converted to wK color layers, right? Even the ground truth segmentations, right? (Those used o be a wK `segmentation` layer...? I guess it does not matter much. The ML can probably read GT from either color or segmentation layers.

Reply:
> Just to clarify, this means all Niftis will be converted to wK color layers, right? Even the ground truth segmentations, right?

Yes! And this has been like this before, so it's not a step back. I intended to auto-detect this and set the correct category, but then ran into problems, because WK does not support multiple segmentation layers. Then, I changed it back to always use color as category, but used the inferred type information to improve the type casting, so that we don't have the 0 vs. 255 issue with segmentation layers.